tests/test_freetextsearch.py

```diff
@@ -313,12 +1619,71 @@ def testBestHitSearchFieldsPostTarget(self):
         self.assertEqual(first_result_ens['approved_symbol'], 'BRAF')
     
         
-        eff_data = json_response['data'][5];
-        self.assertEqual(eff_data['q'], 'eff');
-        self.assertEqual(eff_data['exact'], False);
+        eff_data = json_response['data'][5]
+        self.assertEqual(eff_data['q'], 'eff')
+        self.assertEqual(eff_data['exact'], False)
         self.assertEqual(len(eff_data['data']),1) #should only get name and id
         self.assertEqual(eff_data['data']['approved_symbol'],'UBE2D3')
-        
+
+    def testBestHitSearchExactMatchManyTargetPost(self):
+        targets = GENE_SYMBOL_LIST
+        response = self._make_request('/api/latest/private/besthitsearch',
+                                      data=json.dumps({
+                                          'q': targets,
+                                          'filter': 'target',
+                                          'no_cache': True}),
+                                      content_type='application/json',
+                                      method='POST',
+                                      token=self._AUTO_GET_TOKEN)
+
+        self.assertTrue(response.status_code == 200)
+        json_response = json.loads(response.data.decode('utf-8'))
+
+        self.assertEqual(len(json_response['data']), len(targets))
+
+        for result in json_response['data']:
+            self.assertIsNotNone(result)
+
+    def testBestHitSearchExactMatchMart410TargetPost(self):
+        targets = TEST_MART_410
+        response = self._make_request('/api/latest/private/besthitsearch',
+                                      data=json.dumps({
+                                          'q': targets,
+                                          'filter': 'target',
+                                          'no_cache': True,
+                                      },),
+                                      content_type='application/json',
+                                      method='POST',
+                                      token=self._AUTO_GET_TOKEN)
+
+        self.assertTrue(response.status_code == 200)
+        json_response = json.loads(response.data.decode('utf-8'))
+
+        self.assertEqual(len(json_response['data']), len(targets))
+
+        for result in json_response['data']:
+            self.assertIsNotNone(result)
+
```

Comment:
I think we need to add couple of random checks for the contents of the result - make sure that it does indeed return values expected for specific targets - one or two in the middle somewhere. In previous version when I was testing, I was getting results back, but they had fails inside, for the targets that were working fine in smaller lists. 
so insert bra somewhere in spot 403 and test that it has hits.